### ISSUE TITLE
Error installing vktrs

### ISSUE THREAD

**Issue by runshouse** [dmarx/video-killed-the-radio-star#136]
When I run this:

!pip install vktrs

I get this:

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
Collecting vktrs
  Using cached vktrs-0.1.8-py3-none-any.whl (11 kB)
Requirement already satisfied: scipy in /usr/local/lib/python3.10/dist-packages (from vktrs) (1.10.1)
Collecting python-tsp
  Using cached python_tsp-0.3.1-py3-none-any.whl (18 kB)
Requirement already satisfied: pandas in /usr/local/lib/python3.10/dist-packages (from vktrs) (1.5.3)
Requirement already satisfied: numpy in /usr/local/lib/python3.10/dist-packages (from vktrs) (1.22.4)
Requirement already satisfied: lxml in /usr/local/lib/python3.10/dist-packages (from vktrs) (4.9.2)
Collecting webvtt-py
  Using cached webvtt_py-0.4.6-py3-none-any.whl (16 kB)
Requirement already satisfied: omegaconf in /usr/local/lib/python3.10/dist-packages (from vktrs) (2.3.0)
Requirement already satisfied: toolz in /usr/local/lib/python3.10/dist-packages (from vktrs) (0.12.0)
Collecting yt-dlp
  Using cached yt_dlp-2023.3.4-py2.py3-none-any.whl (2.9 MB)
Collecting pytokenizations
  Using cached pytokenizations-0.8.4.tar.gz (3.8 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  Preparing metadata (pyproject.toml) ... error
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details

**Comment by yyahav** [comment]
I can confirm this issue.
I tried installing older versions but the same error appears.

When trying to build unreleased by:
```
!pip install --upgrade setuptools build
!git clone --branch hf https://github.com/dmarx/video-killed-the-radio-star/
!cd video-killed-the-radio-star;  python -m build; python -m pip install .[api,hf]
```
branch 'hf' does not exist, and when I clone master, I get the same error as OP posted when building.

**Comment by yyahav** [comment]
@dmarx I've found that this issue is not directly related to `vktrs`, it comes from failed building of `pytokenizations`, when `cargo` (rust) is missing, which indeed is missing on Colab which in turn causes this error while installing `pytokenizations`:
```
maturin failed
  Caused by: Cargo metadata failed. Do you have cargo in your PATH?
  Caused by: No such file or directory (os error 2)
```
It seems that `pytokenizations` cannot be installed on Colab, still looking into that

**Comment by yyahav** [comment]
This issue can be closed once #138 is merged.

For the time being if anyone wants to solve it, in the `Installations` cell add the following line:
`!apt-get install cargo`
Just before:
`!pip install vktrs[api,hf]`
